core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/CoreDocumentSynchronizationConfig.java

```diff
@@ -387,13 +387,13 @@ public boolean hasCommittedVersion(final DocumentVersionInfo versionInfo) {
         break;
       case DELETE:
         switch (newestChangeEvent.getOperationType()) {
-          // Coalesce inserts to updates since we believe at some point a document existed remotely
-          // and that this insert should really be an update if we are still in an uncommitted
-          // state.
+          // Coalesce inserts to replaces since we believe at some point a document existed
+          // remotely and that this insert should really be an replace if we are still in an
+          // uncommitted state.
           case INSERT:
             return new ChangeEvent<>(
                 newestChangeEvent.getId(),
-                ChangeEvent.OperationType.UPDATE,
+                ChangeEvent.OperationType.REPLACE,
```

Comment:
Still new at this, but doesn't this have an impact for rules and triggers?

Reply:
Yes but I don't think there's a good solution for this if the user thinks it's really an insert. We can't hold on to enough date for this case.

Reply:
per offline discussion, the answer is that nothing really changed with this particular update since the server was receiving the same events in the end.

your concerns are valid, but they're more because of the limitations of change streams than they are because of the way we've implemented sync